projects/components/src/download-file/service/file-download.service.ts

```diff
@@ -31,25 +31,17 @@ export class FileDownloadService {
    * @param config Csv download config
    */
   public downloadAsCsv(config: CsvDownloadFileConfig): Observable<FileDownloadEvent> {
-    // If given header is empty, then create header from the data keys
-    const header$ = isEmpty(config.header)
-      ? config.dataSource.pipe(
-          map(data => Object.keys(data[0] ?? [])),
-          map(keys => keys.map(startCase))
-        )
-      : of(config.header!);
-
+    const getAllDataKeys = (data: Dictionary<unknown>[]): string[] => uniq(data.map(row => Object.keys(row)).flat());
```

Comment:
nit - now we only do this once, don't need to declare it as a separate function.